### DISCUSSION TITLE
[Bug]: "AI response error; aborting request: Unrecognized algorithm name"

### DISCUSSION THREAD

**Discussion by agambon** [danny-avila/LibreChat#3551] [Troubleshooting]
### What happened?

LibreChat is giving the following error message:
`Something went wrong. Here's the specific error message we encountered: An error occurred while processing your request. Please contact the Admin.
`

The log shows the following:
`
2024-08-05 16:15:17 [31merror[39m: [31m[handleAbortError] AI response error; aborting request: Unrecognized algorithm name[39m`

- LibreChat is running on Docker, v0.7.4-rc1
- No settings have changed recently, and it worked correctly yesterday.
- The API key is correct.
- I am not using a VPN.
- A restart of the Docker container did not fix it.
- Using the chatgpt site directly works correctly still.
- The issue persists regardless of model (4o, 4-turbo-preview, etc)

### Steps to Reproduce

1. Open LibreChat
2. Attempt to send a message
3. Error returned

### What browsers are you seeing the problem on?

Chrome

### Relevant log output

```shell
2024/08/05 12:07:29	stdout	[32m  "user": "661eacc577e657c18e7e4ad4"[39m
2024/08/05 12:07:29	stdout	[32m  "isCreatedByUser": false,[39m
2024/08/05 12:07:29	stdout	[32m  "text": "An error occurred while processing your request. Please contact the Admin.",[39m
2024/08/05 12:07:29	stdout	[32m  "final": true,[39m
2024/08/05 12:07:29	stdout	[32m  "error": true,[39m
2024/08/05 12:07:29	stdout	[32m  "unfinished": false,[39m
2024/08/05 12:07:29	stdout	[32m  "parentMessageId": "00000000-0000-0000-0000-000000000000",[39m
2024/08/05 12:07:29	stdout	[32m  "conversationId": null,[39m
2024/08/05 12:07:29	stdout	[32m  "messageId": "019c3b91-29fd-4050-9ba5-0253557099cc",[39m
2024/08/05 12:07:29	stdout	[32m  "sender": "GPT-4",[39m
2024/08/05 12:07:29	stdout	[32m{[39m
2024/08/05 12:07:29	stdout	
2024/08/05 12:07:29	stdout	2024-08-05 16:07:29 [32minfo[39m: [32m---Invalid conversation ID Params:[39m
2024/08/05 12:07:29	stdout	2024-08-05 16:07:29 [33m[3mwarn[23m[39m: [33m[3mInvalid conversation ID: null[23m[39m
2024/08/05 12:07:29	stdout	2024-08-05 16:07:29 [31merror[39m: [31m[handleAbortError] AI response error; aborting request: Unrecognized algorithm name[39m
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
There isn't enough context/info here to properly troubleshoot. What endpoint/model, what are your configurations (reverse proxy, etc)?

Personally, I've never seen this error, and as you can see, it's being thrown by the API, not LibreChat

**Comment by agambon** [comment]
The issue was found when trying to use OpenAI's gpt-4o model, but the issue still occurs when using other OpenAI models: gpt-4-turbo and gpt-4-turbo-preview. 

I tried to test using Anthropic/Claude. My Anthropic key was expired, so I put in the key. The docker container then crashed and restarted, with the following log:

```
2024/08/05 12:41:40	stdout	2024-08-05 16:41:40 [33m[3mwarn[23m[39m: [33m[3mDefault value for JWT_REFRESH_SECRET is being used.[23m[39m
2024/08/05 12:41:40	stdout	2024-08-05 16:41:40 [33m[3mwarn[23m[39m: [33m[3mDefault value for JWT_SECRET is being used.[23m[39m
2024/08/05 12:41:40	stdout	2024-08-05 16:41:40 [33m[3mwarn[23m[39m: [33m[3mDefault value for CREDS_IV is being used.[23m[39m
2024/08/05 12:41:40	stdout	2024-08-05 16:41:40 [33m[3mwarn[23m[39m: [33m[3mDefault value for CREDS_KEY is being used.[23m[39m
2024/08/05 12:41:40	stdout	2024-08-05 16:41:40 [31merror[39m: [31mConfig file YAML format is invalid: ENOENT: no such file or directory, open '/app/librechat.yaml'[39m
2024/08/05 12:41:39	stdout	2024-08-05 16:41:39 [32minfo[39m: [32mConnected to MongoDB[39m
2024/08/05 12:41:31	stdout	2024-08-05 16:41:31 [32minfo[39m: [32m[Optional] Redis not initialized. Note: Redis support is experimental.[39m
2024/08/05 12:41:27	stdout	
2024/08/05 12:41:27	stdout	> cross-env NODE_ENV=production node api/server/index.js
2024/08/05 12:41:27	stdout	> [email protected] backend
2024/08/05 12:41:27	stdout	
2024/08/05 12:41:17	stderr	npm notice
2024/08/05 12:41:17	stderr	npm notice To update run: npm install -g [email protected]
2024/08/05 12:41:17	stderr	npm notice Changelog: https://github.com/npm/cli/releases/tag/v10.8.2
2024/08/05 12:41:17	stderr	npm notice New patch version of npm available! 10.8.1 -> 10.8.2
2024/08/05 12:41:17	stderr	npm notice
2024/08/05 12:41:16	stdout	2024-08-05 16:41:16 [31merror[39m: [31mThere was an uncaught error: Unrecognized algorithm name[39m
2024/08/05 12:41:10	stdout	[32m[39m
2024/08/05 12:41:10	stdout	
2024/08/05 12:41:10	stdout	[32m}[39m
2024/08/05 12:41:10	stdout	[32m  "user": "661eacc577e657c18e7e4ad4"[39m
2024/08/05 12:41:10	stdout	[32m  "isCreatedByUser": false,[39m
2024/08/05 12:41:10	stdout	[32m  "text": "{\"type\":\"no_user_key\"}",[39m
2024/08/05 12:41:10	stdout	[32m  "final": true,[39m
2024/08/05 12:41:10	stdout	[32m  "error": true,[39m
2024/08/05 12:41:10	stdout	[32m  "unfinished": false,[39m
2024/08/05 12:41:10	stdout	[32m  "parentMessageId": "00000000-0000-0000-0000-000000000000",[39m
2024/08/05 12:41:10	stdout	[32m  "conversationId": null,[39m
2024/08/05 12:41:10	stdout	[32m  "messageId": "ab664e14-d127-4752-8e7a-73634ee2b730",[39m
2024/08/05 12:41:10	stdout	[32m  "sender": "Claude",[39m
```

Odd behavior, as I have used an anthropic key successfully in the past.

On reload, the Anthropic key is still missing, and attempts to add the Anthropic key still result in crashing.

Not sure if or how these two issues could be connected. The ChatGPT API won't cooperate, and the program crashes when trying to add an Anthropic key for testing. 

My network configuration is standard: IP assigned by ISP, PC behind router.

**Comment by agambon** [comment]
Is `Unrecognized algorithm name` a response returned by the chatgpt API, or by LibreChat?

**Comment by agambon** [comment]
The openai api key hasn't changed, but I thought I'd troubleshoot by re-entering my openai api key.

Docker container then crashes with the following log:
`2024/08/05 13:30:27	stdout	2024-08-05 17:30:27 [31merror[39m: [31mThere was an uncaught error: Unrecognized algorithm name[39m`

When it reloads, it seems like the api key is retained, but the original reported misbehavior persists.

Strange, and I can't make heads or tails of it.

**Comment by agambon** [comment]
Continuing to troubleshoot here, hoping that additional info can help resolve this:
Attempting use groq or Mistral (have never used them before) results in the same error.
`Something went wrong. Here's the specific error message we encountered: An error occurred while processing your request. Please contact the Admin.`

Seems to be something borked with my librechat install all of a sudden.

The log says `Unrecognized algorithm name` but also says `Invalid conversation ID`. I'm unsure if either is meaningful in tracking down the issue.

**Reply by danny-avila** [reply]
@agambon thanks for reporting with your logs, I started looking into it and looks like this was introduced yesterday here: https://github.com/danny-avila/LibreChat/pull/3357

Seems odd this is only affecting use of `user_provided` API keys as the methods are used elsewhere pretty frequently. I'm also testing a fix at the moment.

**Comment by agambon** [comment]
Thanks for looking into it! Happy to wait for a fix. Cheers!

**Reply by danny-avila** [reply]
you're welcome! Just merged the fix, let me know if it works for you, you may need to wait for the docker images to build if you use docker.

https://github.com/danny-avila/LibreChat/pull/3556

**Comment by agambon** [comment]
The issue is fixed! Thanks, will close this now.
